### PR TITLE
Issues 2104 2114 2116 2117 2118 42 stable

### PR DESCRIPTION
**Pull request by JustinKyleJames**
This is a cherry-pick of commits that have been merged to 4.3 with some necessary changes to meet 4.2 API's.

### CODE REVIEW

File: packaging/resource_suite_s3_nocache.py [hunk 1]
```diff
@@ -2277,6 +2132,195 @@ def test_s3_in_replication_node__issue_2102(self):
             self.admin.assert_icommand("iadmin rmresc %s" % s3_resc2_resource_name)
             self.admin.assert_icommand("iadmin rmresc %s" % s3_resc3_resource_name)
 
+# The tests in this class take a long time and do not need to run in every different test suite
+class Test_S3_NoCache_Large_File_Tests_Base(Test_S3_NoCache_Base):
```
Comment (korydraughn) [inline packaging/resource_suite_s3_nocache.py:2136]: The tests in red above are the ones you moved into this class, correct?

**Review by korydraughn — COMMENTED** [review]
Has this PR passed all tests?
Is there anything you want us to focus on in particular?

**Comment by alanking** [comment]
Cherry-picked from https://github.com/irods/irods_resource_plugin_s3/pull/2120
Cherry-picked from https://github.com/irods/irods_resource_plugin_s3/pull/2116

@JustinKyleJames - Please confirm

File: packaging/resource_suite_s3_nocache.py (hunk 1 above)
Comment (JustinKyleJames) [reply packaging/resource_suite_s3_nocache.py:2136]: Yes, there are three tests that were moved.

**Comment by JustinKyleJames** [comment]
> sed all tests?
> Is there anything you want us to focus on in particular

Yes, all tests passed.

**Comment by JustinKyleJames** [comment]
> Cherry-picked from #2120 Cherry-picked from #2116
> 
> @JustinKyleJames - Please confirm

Yes, that is correct.

File: packaging/resource_suite_s3_nocache.py (hunk 1 above)
Comment (korydraughn) [reply packaging/resource_suite_s3_nocache.py:2136]: Cool. You can mark this as resolved.

**Comment by JustinKyleJames** [comment]
> Everything looked the same as main except the logging stuff, so I think this is good to go. Please add #s.

Done

**Pull request merged**